### DISCUSSION TITLE
Issue deploy Flutter Web to Firebase Hosting

### DISCUSSION THREAD

**Discussion by kevin-induro** [codemagic-ci-cd/codemagic-docs#1980] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I attempted to deploy my Flutter web project to Firebase Hosting using the following commands
```
firebase experiments:enable webframeworks
firebase deploy --project $GCP_PROJECT_ID --token $FIREBASE_TOKEN
```

### Expected results

I expected the deployment to work.

### Actual results

Instead, I got an error stating: `Could not determine the web framework in use.`

This was a reported issue for the Firebase CLI: https://github.com/firebase/firebase-tools/issues/6085
Updating the build machine's version of firebase_tools to [12.4.5](https://firebase.google.com/support/release-notes/cli#version_1245_-_july_19_2023) will fix this issue. For now, I run npm i the tools in my script to get the latest version, but that's brittle and slow.

**Comment by kulakarla** [comment]
Hello @kevin-induro 

Looking at the issue, it seems the bug could indeed be tied to versions, but in the thread you sent us, people fixed it with different specific versions. Therefore we don't know which version would be an actual fix. In addition, some user solved it with a different approach.

It's good that you got it working and we'll keep our eye on it.

**Reply by kevin-induro** [reply]
Thanks for looking into it! Perhaps you missed the link to version 12.4.5 where they document that this specific issue is fixed.

**Reply by kulakarla** [reply]
Thank you for letting us know. We'll investigate it a bit and see what we'll proceed with. Currently, please continue using your custom script as updating software on our side requires some consideration.

If you could post your installation script here, it would be greatly appreciated. I would select it as an answer.

**Reply by kevin-induro** [reply, marked as answer]
My current script looks like this:
```
    - &build_web
      name: Build web
      # remove install once firebase-tools is version 12.4.5
      script: | 
        npm install -g firebase-tools
        firebase experiments:enable webframeworks
        firebase deploy --project $GCP_PROJECT_ID
```
